src/dp_port.c

```diff
@@ -458,3 +468,63 @@ int dp_stop_port(struct dp_port *port)
 	port->allocated = false;
 	return DP_OK;
 }
+
+static int dp_port_total_flow_meter_config(struct dp_port *port, uint64_t total_flow_rate_cap)
+{
+	return dp_set_vf_rate_limit(port->port_id, total_flow_rate_cap);
+}
+
+static int dp_port_public_flow_meter_config(struct dp_port *port, uint64_t public_flow_rate_cap)
+{
+	struct rte_meter_srtcm_params srtcm_params = dp_srtcm_params_base;
+	int ret;
+
+	srtcm_params.cir = DP_METER_CIR_BASE_VALUE * (public_flow_rate_cap / 8); // Mbits/s -> bytes/s
+	if (public_flow_rate_cap < DP_METER_EBS_BREAK_VALUE)
+		srtcm_params.ebs = public_flow_rate_cap * DP_METER_MBITS_TO_BYTES;
+
+	ret = rte_meter_srtcm_profile_config(&port->port_srtcm_profile, &srtcm_params);
+	if (DP_FAILED(ret)) {
+		DPS_LOG_ERR("Cannot configure meter profile", DP_LOG_PORT(port), DP_LOG_RET(ret));
+		return DP_ERROR;
+	}
+
+	ret = rte_meter_srtcm_config(&port->port_srtcm, &port->port_srtcm_profile);
+	if (DP_FAILED(ret)) {
+		DPS_LOG_ERR("Cannot configure meter", DP_LOG_PORT(port), DP_LOG_RET(ret));
+		return DP_ERROR;
```

Comment:
Now this may not be necessary (you will know the answer to this), but just for completeness, the error branch should reset the srtcm_profile structure from previous block.

Reply:
indeed it was not necessary as it is a value assignment and a decision value will take effect if assighment failed here. But for the completeness, the struct in the previous block is set to 0 to erase the assigned values